content/concepts/multiplex/early-negotiation.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "Early Multiplexer Negotiation"
+description: "Peers can use security protocol extenstions for early muxer negotiation."
+weight: 162
+---
+
+## Typical connection upgrade process
+
+Peers upgrade raw transport connections by using the same
+[multistream-selection](https://github.com/multiformats/multistream-select)
+protocol to negotiate security and stream multiplexing.
+
+First, security is established, and a security handshake is performed
+either for [Noise](../secure-comm/noise) or [TLS 1.3](../secure-comm/tls).
+Multistream-select will run again on top of
+the secure channel to negotiate a steam muxer, like [yamux](yamux).
+
+<!-- ADD DIAGRAM -->
```

Comment:
Are we planning to merge without resolving this TODO? Should we create an issue so we don't forget?

Reply:
I have them noted down, and will also add to the issue open about unified diagrams.